vcd/resource_vcd_org_group.go

```diff
@@ -0,0 +1,192 @@
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true, // vCD does not allow to change provider type
+				Description:  "SAML group name",
+				ValidateFunc: validation.StringInSlice([]string{"SAML", "INTEGRATED"}, false),
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description",
+			},
+			"role": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Existing role name to assign",
+			},
+		},
+	}
+}
+
+func resourceVcdOrgGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	roleName := d.Get("role").(string)
+	role, err := adminOrg.GetRoleReference(roleName)
+	if err != nil {
+		return fmt.Errorf("unable to find role %s: %s", roleName, err)
+	}
+
+	newGroup := govcd.NewGroup(&vcdClient.Client, adminOrg)
+	groupDefinition := types.Group{
+		Name:         d.Get("name").(string),
+		Role:         role,
+		ProviderType: d.Get("provider_type").(string),
+		Description:  d.Get("description").(string),
+	}
+	newGroup.Group = &groupDefinition
+
+	createdGroup, err := adminOrg.CreateGroup(newGroup.Group)
+	if err != nil {
+		return fmt.Errorf("error creating group %s: %s", groupDefinition.Name, err)
+	}
+
+	d.SetId(createdGroup.Group.ID)
+
+	return resourceVcdOrgGroupRead(d, meta)
+}
+
+func resourceVcdOrgGroupRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	group, err := adminOrg.GetGroupById(d.Id(), false)
+	if govcd.IsNotFound(err) {
+		log.Printf("error finding group for deletion %s: %s. Removing from state", d.Id(), err)
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error finding group for deletion  %s: %s", d.Id(), err)
+	}
+
+	d.Set("name", group.Group.Name)
+	d.Set("description", group.Group.Description)
+	d.Set("role", group.Group.Role.Name)
+	d.Set("provider_type", group.Group.ProviderType)
+
+	return nil
+}
+
+func resourceVcdOrgGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	group, err := adminOrg.GetGroupById(d.Id(), false)
+	if err != nil {
+		return fmt.Errorf("error finding group for update %s: %s", group.Group.Name, err)
+	}
+
+	// Role change
+	if d.HasChange("role") {
+		roleName := d.Get("role").(string)
+		role, err := adminOrg.GetRoleReference(roleName)
+		if err != nil {
+			return fmt.Errorf("unable to find role %s: %s", roleName, err)
+		}
+		group.Group.Role = role
+	}
+
+	// Description change
+	if d.HasChange("description") {
```

Comment:
Could you add a note, that works only for SAML ?

Reply:
Set